src/fe-common/core/fe-common-core.c

```diff
@@ -470,16 +470,23 @@ gboolean strarray_find_dest(char **array, const TEXT_DEST_REC *dest)
 		return TRUE;
 
 	if (dest->server_tag != NULL) {
-		char *tagtarget = g_strdup_printf("%s/%s", dest->server_tag, "*");
+		char *tagtarget = g_strdup_printf("%s/%s", dest->server_tag, "#");
```

Comment:
All the entries in the list for a given tag start with `tagname/` so it makes sense to do a prefix search on the list and then check whether the part after the slash is a `@`,`#`,`*` or a valid `target` name.
One more detail jumped to my mind, if the user sets the list to `Freenode/@,Freenode/#` should we respect both even though that's better expressed as `Freenode/*`?